src/mutations/item.test.ts

```diff
@@ -0,0 +1,214 @@
+import { act } from '@testing-library/react-hooks';
+import nock from 'nock';
+import { Map, Record } from 'immutable';
+import { buildCopyItemRoute, buildCopyItemsRoute, buildMoveItemRoute, buildMoveItemsRoute } from '../api/routes';
+import { setUpTest, mockMutation, waitForMutation } from '../../test/utils';
+import { REQUEST_METHODS } from '../api/utils';
+import {
+  OK_RESPONSE,
+  ITEMS
+} from '../../test/constants';
+import {
+  buildItemKey,
+  MUTATION_KEYS
+} from '../config/keys';
+import { Item } from '../types';
+
+
+const { wrapper, queryClient, useMutation } = setUpTest();
+describe('items Mutations', () => {
+  afterEach(() => {
+    queryClient.clear();
+    nock.cleanAll();
+  });
+
+  describe(MUTATION_KEYS.COPY_ITEM, () => {
+    const to = ITEMS[0].id;
+    const copied = ITEMS[1].id;
+
+    const route = `/${buildCopyItemRoute(copied)}`;
+    const mutation = () => useMutation(MUTATION_KEYS.COPY_ITEM);
+
+    // set random data in cache
+    ITEMS.forEach(item => {
+      const itemKey = buildItemKey(item.id);
+      queryClient.setQueryData(itemKey, Map(item));
+    });
+
+    it('singleCopySuccess', async () => {
+      const response = OK_RESPONSE;
+
+      const endpoints = [
+        {
+          response,
+          method: REQUEST_METHODS.POST,
+          route,
+        }
+      ];
+
+      const mockedMutation = await mockMutation({
+        endpoints,
+        mutation,
+        wrapper,
+      });
+
+      await act(async () => {
+        await mockedMutation.mutate({
+          to,
+          id: copied,
+        });
+        await waitForMutation();
+      });
+
+      // verify cache keys
+      const itemKey = buildItemKey(to);
+      expect(queryClient.getQueryState(itemKey)?.isInvalidated).toBeFalsy();
```

Comment:
If the copy happens, shouldn't the `to` item be invalidated? As you've seen, previous code can contain bugs..

Additionally, I would check that the item you are copying is still valid/contains the exact same item.